client.ts

```diff
@@ -0,0 +1,463 @@
+    const response = authorizationModelsResponse as any as CallResult<ReadAuthorizationModelResponse>;
+    response.authorization_model = authorizationModelsResponse.authorization_models?.[0];
+    delete (response as any).authorization_models;
+    return response;
+  }
+
+  /***********************
+   * Relationship Tuples *
+   ***********************/
+
+  /**
+   * Read Changes - Reads the list of historical tuple writes and deletes
+   * @param clientReadChangesRequest
+   * @param {object} paginationOptions
+   * @param {number} [paginationOptions.pageSize]
+   * @param {string} [paginationOptions.continuationToken]
+   * @param options
+   */
+  async readChanges(clientReadChangesRequest: ClientReadChangesRequest, paginationOptions: PaginationOptions = {}, options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ReadChangesResponse> {
+    return this.openFgaApi.readChanges(clientReadChangesRequest.type, paginationOptions.pageSize, paginationOptions.continuationToken, options);
+  }
+
+  /**
+   * Read - Reads tuples previously written to the store (does not evaluate)
+   * @param clientReadRequest
+   * @param options
+   */
+  async read(clientReadRequest: ClientReadRequest, options: ClientRequestOpts = {}): PromiseResult<ReadResponse> {
+    return this.openFgaApi.read({ tuple_key: clientReadRequest }, options);
+  }
+
+  /**
+   * Write - Create or delete relationship tuples
+   * @param {object} body
+   * @param {objects} options
+   */
+  async write(body: ClientWriteRequest, options: ClientRequestOptsWithAuthZModelId & ClientWriteRequestOpts = {}): Promise<ClientWriteResponse> {
+    const { transaction = {} } = options;
+    const { writes, deletes } = body;
+    const authorizationModelId = this.getAuthorizationModelId(options);
+
+    if (!transaction || transaction.disable) {
+      await this.openFgaApi.write({
+        writes: { tuple_keys: writes || [] },
+        deletes: { tuple_keys: deletes || [] },
+        authorization_model_id: authorizationModelId,
+      }, options);
+      return {
+        writes: writes?.map(tuple => ({
+          tuple_key: tuple,
+          status: ClientWriteStatus.SUCCESS,
+        })) || [],
+        deletes: deletes?.map(tuple => ({
+          tuple_key: tuple,
+          status: ClientWriteStatus.SUCCESS,
+        })) || []
+      };
+    }
+
+    const results: ClientWriteResponse = { writes: [], deletes: [] };
+    await chunkSequentialCall<TupleKey, void>(
+      (chunk) => this.openFgaApi.write({ writes: { tuple_keys: chunk}, authorization_model_id: authorizationModelId })
+        .then(() => { results.writes.push(...chunk.map(tuple => ({ tuple_key: tuple, status: ClientWriteStatus.SUCCESS }))); })
+        .catch(() => { results.writes.push(...chunk.map(tuple => ({ tuple_key: tuple, status: ClientWriteStatus.FAILURE }))); }),
+      writes || [],
+      transaction.maxPerChunk || DEFAULT_MAX_METHOD_PARALLEL_REQS,
+    );
```

Comment:
we will need to add some form of sleep/retry in case of 429 mechanism in between the calls.  Otherwise, the caller may get rate limited in some OpenFGA implementations.